### PR TITLE
File rename support.

### PR DESCRIPTION
**Pull request by sa-bpelakh**
To support the functionality requested by Jamie.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -165,6 +165,15 @@ one of the following values:
...
+  - `rename` - If provided, must contain `from` and `to` attributes. When specified, each file
+    is renamed as it is copied, where `from` is treated as a Python regular expression, and `to`
+    as the substitution string. Backreferences are available for capturing groups, e.g.
+    ```yaml
+      rename:
+        from: "(.*)\\.owl"
+        to: "\\g<1>{version}.owl"
+    ```
+    will add a version number to the base name of each `.owl` file.
   - `replace` - If provided, must contain `from` and `to` attributes. When specified, each file
```
Comment (rjyounes) [inline README.md:177]: Make the difference between replace and rename explicit - I.e., that replace is for text in files.

File: README.md [hunk 2]
```diff
@@ -165,6 +165,15 @@ one of the following values:
...
     is processed after being copied, and each instance of the `from` string is replaced with `to`.
```
Comment (rjyounes) [inline README.md:178]: Add same notes about regex and backreferences as in rename

File: README.md [hunk 3]
```diff
@@ -165,6 +165,15 @@ one of the following values:
     assumed to be file paths to a single file. If `includes` is provided, `source` and `target` are 
     assumed to be directories, and each member of the `includes` list a glob pattern inside the
     `source` directory.
+  - `rename` - If provided, must contain `from` and `to` attributes. When specified, each file
+    is renamed as it is copied, where `from` is treated as a Python regular expression, and `to`
+    as the substitution string. Backreferences are available for capturing groups, e.g.
+    ```yaml
+      rename:
+        from: "(.*)\\.owl"
+        to: "\\g<1>{version}.owl"
```
Comment (rjyounes) [inline README.md:174]: This syntax for the replace is very unfamiliar. Maybe add a link to the documentation?

File: README.md (hunk 1 above)
Comment (sa-bpelakh) [reply README.md:180]: Specified 'file contents' for `replace`.

File: README.md (hunk 3 above)
Comment (sa-bpelakh) [reply README.md:175]: Included link to Python regex replace docs.

File: README.md (hunk 2 above)
Comment (sa-bpelakh) [reply README.md]: :+1:

**Pull request merged**